sdk/data_kusto/src/error.rs

```diff
@@ -0,0 +1,41 @@
+//! Defines `KustoRsError` for representing failures in various operations.
+use std::fmt::Debug;
+use thiserror;
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
```

Comment:
Just as an FYI: we want to move away from crates have their own error types in favor of just a single error type defined in azure_core. This is fine for now though - we can make that change later once we finish making the change in the cosmos crate.